tests/test_coroutine_cached_property.py

```diff
@@ -1,125 +1,102 @@
-# -*- coding: utf-8 -*-
 """
 The same tests as in :mod:`.test_async_cached_property`, but with the old
 yield from instead of the new async/await syntax. Used to test Python 3.4
 compatibility which has asyncio but doesn't have async/await yet.
 """
 
 import unittest
-import asyncio
 from freezegun import freeze_time
 
 import cached_property
 
 
-def unittest_run_loop(f):
-    def wrapper(*args, **kwargs):
-        coro = asyncio.coroutine(f)
-        future = coro(*args, **kwargs)
-        loop = asyncio.get_event_loop()
-        loop.run_until_complete(future)
-
-    return wrapper
-
-
 def CheckFactory(cached_property_decorator):
     """
     Create dynamically a Check class whose add_cached method is decorated by
     the cached_property_decorator.
     """
 
-    class Check(object):
+    class Check:
         def __init__(self):
             self.control_total = 0
             self.cached_total = 0
 
-        @asyncio.coroutine
-        def add_control(self):
+        async def add_control(self):
             self.control_total += 1
             return self.control_total
 
         @cached_property_decorator
-        @asyncio.coroutine
-        def add_cached(self):
+        async def add_cached(self):
             self.cached_total += 1
             return self.cached_total
 
     return Check
 
 
-class TestCachedProperty(unittest.TestCase):
+class TestCachedProperty(unittest.IsolatedAsyncioTestCase):
     """Tests for cached_property"""
 
     cached_property_factory = cached_property.cached_property
 
-    @asyncio.coroutine
-    def assert_control(self, check, expected):
+    async def assert_control(self, check, expected):
         """
         Assert that both `add_control` and 'control_total` equal `expected`
         """
-        value = yield from check.add_control()  # noqa
-        self.assertEqual(value, expected)
```

Comment:
I exchanged these as it is confusing if in the error output the actual and expected values are swapped - and I got quite a few failures on the first try...